src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/services/TestGenieTelemetryService.kt

```diff
@@ -28,27 +30,60 @@ class TestGenieTelemetryService {
     }
 
     /**
-     * Sends the telemetry to the TestGenie server.
+     * Saves the telemetry to a file.
      */
-    fun uploadTelemetry() {
+    fun submitTelemetry() {
         if (!telemetryEnabled) {
             return
         }
 
-        val testCasesToUpload = mutableListOf<ModifiedTestCase>()
+        val testCasesToSubmit = mutableListOf<ModifiedTestCase>()
 
         synchronized(modifiedTestCasesLock) {
-            testCasesToUpload.addAll(modifiedTestCases)
+            testCasesToSubmit.addAll(modifiedTestCases)
             modifiedTestCases.clear()
         }
 
-        log.info("Uploading ${testCasesToUpload.size} test cases to server")
+        // If there are no tests to submit, do not create a file
+        if (testCasesToSubmit.size == 0) {
+            return
+        }
+
+        log.info("Submiting ${testCasesToSubmit.size} test cases to a file")
 
         val gson = Gson()
-        val json = gson.toJson(testCasesToUpload)
-        log.info("Uploading test cases: $json")
+        val json = gson.toJson(testCasesToSubmit)
+        log.info("Submiting test cases: $json")
+
+        writeTelemetryToFile(json)
+    }
+
+    /**
+     * Writes a json with the telemetry to a file.
+     *
+     * @param json a json object with the telemetry
+     */
+    private fun writeTelemetryToFile(json: String) {
+        // Get the separator depending on the underlying OS
+        val separator: String = if (System.getProperty("os.name").contains("Windows")) "\\" else "/"
+        // Get the telemetry path
+        var dirName: String = TestGenieSettingsService.getInstance().state?.telemetryPath
+            ?: System.getProperty("user.dir")
```

Comment:
Maybe the default path should be something like `<user.dir>/TestGenieTelemetry` instead of just dumping files in the home directory?

Reply:
Sure, just leave it as is then.